### ISSUE TITLE
Arango not starting after OOM crash (24 hours and counting)

### ISSUE THREAD

**Issue by aveiros** [arangodb/arangodb#12299]
## My Environment

* __ArangoDB Version__:        3.5.3
* __Storage Engine__:             RocksDB
* __Deployment Mode__:       Active Failover
* __Deployment Strategy__:   Manual Start
* __Configuration__:               (below)
* __Infrastructure__:               own
* __Operating System__:        Red Hat Enterprise Linux Server release 7.8
* __Total RAM in your machine__:        376 GB
* __Disks in use__:        SSD
* __Used Package__:              rpm

## Component, Query & Data

__Affected feature__:
n/a

__AQL query (if applicable)__:
n/a

__AQL explain (if applicable)__:
n/a

__Dataset__:
not sure, as the crash occurred before we actually saw the numbers. on Friday we had 145 million docs in one of the collections.

__Size of your Dataset on disk__:
3 TB (2.8 TB is engine-rocksdb)

__Replication Factor & Number of Shards (Cluster only)__:
n/a

__Other information__:
**CPUs**
```
Total cpus: 40
```

**Memory usage (free -m)**
```
Mem: total 376 GB, udf 310GB, free 5.5GB shared 224M buff/cache 60G available 64G
Swap total 8.0G used 401M free 7.6G
```

**DB Setup, running in Active Failover with 1 Agency and 1 Leader***
*Agency Startup Command*

```
arangod --server.authentication false --server.endpoint tcp://0.0.0.0:8519 --agency.my-address tcp://<IP>:8519 --agency.activate true --agency-supervision true --agency-size 1 --database-directory /data/failover/agency --log.role true --log.level debug --log.use-local-time true --log.output file://data/log/agency.log
```

*Leader Startup Command*

```
arangod --server-authentication false --server.endpoint tcp://0.0.0.0:8529 --cluster-my-address tcp://<IP>:8529 --cluster.my-role SINGLE --cluster.agency-endpoint tcp://<IP>:8519 --log-role true --log.level debug --log.use-local-time true --log.output file://data/log/dbserver.log --database-directory /data/failover/dbserver --replication.automatic-failover true
```

**/etc/sysctl.conf**
```
# when changing the fs-max make sure it's greater than nr_open, see https://access.redhat.com/solutions/43926
vm.max_map_count=8192000
fs.file-max=10000000
fs.nr_open=20000000
```

**/etc/security/limits.conf**
```
arangodb soft nofile 10000000
arangodb hard nofile 10000000
arangodb soft nproc  10000000
arangodb hard nproc  10000000
```

**cat /sys/kernel/mm/transparent_hugepage/enabled**
```
always [madvise] never
```

**cat /sys/kernel/mm/transparent_hugepage/defrag**
```
always [madvise] never
```

**cat /proc/sys/vm/overcommit_memory**
```
0
```

**cat /proc/sys/vm/zone_reclaim_mode**
```
0
```

We have **2 collections**, each one of the collections has a associated View with a custom Analyzer as follow
View
```
{
  "links": {
    "entities": {
      "analyzers": [
        "identity",
        "text_standard"
      ],
      "includeAllFields": true,
      "storeValues": "id",
      "trackListPositions": false
    }
  }
}
```

Analyzer
```
{
  "name": "text_standard",
  "type": "text",
  "properties": {
    "locale": "XX.utf-8",
    "case": "lower",
    "stopwords": [],
    "accent": false,
    "stemming": false
  },
  "features": [
    "position",
    "norm",
    "frequency"
  ]
}
```

## Where we at currently
1. We have a primary database containing ~ *collection 1* => 560 million docs and *collection 2* => 288 million docs
2. Since Thursday (23rd), we are ingesting data from the primary database into ArangoDB (using Apache Nifi).
3. Yesterday morning ArangoDB experienced a OOM crash around 9AM
4. We restarted Arango and since then it hasn't come up yet (+24 hours later).

Can you provide us some guidance on this?

**Comment by Simran-B** [comment]
Internal reference: https://arangodb.atlassian.net/browse/ES-662

**Comment by maxkernbach** [comment]
Fixed with 3.7.5.
